build-system/eslint-rules/no-non-string-log-args.js

```diff
@@ -0,0 +1,131 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @typedef {{
+ *   name: string.
+ *   variadle: boolean,
+ *   startPos: number
+ * }}
+ */
+const LogMethodMetadata = null;
+
+/**
+ * @type {!Array<LogMethodMetadata>}
+ */
+const transformableMethods = [
+  {name: 'assert', variadic: false, startPos: 1},
+  {name: 'assertString', variadic: false, startPos: 1},
+  {name: 'assertNumber', variadic: false, startPos: 1},
+  {name: 'assertBoolean', variadic: false, startPos: 1},
+  {name: 'assertEnumValue', variadic: false, startPos: 2},
+  {name: 'assertElement', variadic: false, startPos: 1},
+  {name: 'createExpectedError', variadic: true, startPos: 0},
+  {name: 'fine', variadic: true, startPos: 1},
+  {name: 'info', variadic: true, startPos: 1},
+  {name: 'warn', variadic: true, startPos: 1},
+  {name: 'error', variadic: true, startPos: 1},
+  {name: 'expectedError', variadic: true, startPos: 1},
+  {name: 'createError', variadic: true, startPos: 0},
+];
+
+/**
+ * @param {!Node} node
+ * @return {boolean}
+ */
+function areAllArgumentsLiteral(node) {
+  if (node.type === 'Literal') {
+    return true;
+  }
+  // Allow for string concatenation operations.
+  if (node.type === 'BinaryExpression' && node.operator === '+') {
+    if (areAllArgumentsLiteral(node.left) &&
+        areAllArgumentsLiteral(node.right)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/**
+ * @param {string} name
+ * @return {!LogMethodMetadata}
+ */
+function getMetadata(name) {
+  for (let i = 0; i < transformableMethods.length; i++) {
+    const curTransformableMethod = transformableMethods[i];
+    if (curTransformableMethod.name === name) {
+      return curTransformableMethod;
+    }
+  }
+  return null;
+}
+
+const expressions = transformableMethods.map(x => {
+  return `CallExpression[callee.property.name=${x.name}]`;
+}).join(',');
+
+
+module.exports = function(context) {
+  return {
+    [expressions]: function(node) {
+      // Make sure that callee is a CallExpression as well.
+      // dev().assert() // enforce rule
+      // dev.assert() // ignore
+      if (!(node.callee.object &&
+          node.callee.object.type === 'CallExpression')) {
+        return;
+      }
+
+      // Make sure that the CallExpression is one of dev() or user().
+      if(!['dev', 'user'].includes(node.callee.object.callee.name)) {
+        return;
+      }
+
+      const methodInvokedName = node.callee.property.name;
+      // Find the position of the argument we care about.
+      const metadata = getMetadata(methodInvokedName);
+
+      // If there's no metadata, this is most likely a test file running
+      // private methods on log.
+      if (!metadata) {
+        return;
+      }
+
+      const argToEval = node.arguments[metadata.startPos];
+
+      if (!argToEval) {
```

Comment:
Not every `dev().assert()` has a message.

Reply:
Should every `dev().assert()` have a message?

Reply:
good q. i missed the interaction between the assertion removal code and this message extraction code.

since we remove `dev()  => "assert", "fine", "assertElement", "assertString", "assertNumber", "assertBoolean"`

from the output js files. does it make sense to even extract these messages (if any)

Reply:
> Should every dev().assert() have a message?

No, it's generally used only for typecasting.

> does it make sense to even extract these messages (if any)

No, I would think not.

Reply:
yeah, it wouldnt matter anyways as long as removal happens first.

Reply:
Still need to remove this report.

Reply:
removed. i was thinking i did want to enforce it for non `assert` prefixed methods though